File: magma-platform/magma-platform-bukkit/src/main/java/de/jalumu/magma/platform/bukkit/player/BukkitPlayerProvider.java
package de.jalumu.magma.platform.bukkit.player;

import de.jalumu.magma.platform.base.player.BasePlayerProvider;
import de.jalumu.magma.platform.bukkit.bootstrap.MagmaBukkitBootstrap;
import de.jalumu.magma.player.FetchedPlayer;
import de.jalumu.magma.player.MagmaPlayer;
import de.jalumu.magma.player.PlayerProvider;
import org.bukkit.Bukkit;
import org.bukkit.OfflinePlayer;

import java.util.UUID;

public class BukkitPlayerProvider extends BasePlayerProvider {

    public BukkitPlayerProvider(MagmaBukkitBootstrap bukkitBootstrap) {
        super();
    }

    @Override
    protected FetchedPlayer getCachedPlayer(String name) {
        return fetchPlayer(name);
    }

    @Override
    protected FetchedPlayer getCachedPlayer(UUID uuid) {
        return fetchPlayer(uuid);
    }

    @Override
    protected MagmaPlayer getOnlinePlayer(String name) {
        return null;
    }

    @Override
    protected MagmaPlayer getOnlinePlayer(UUID uuid) {
        return null;
    }

    @Override
    protected MagmaPlayer[] getOnlinePlayers() {
        return new MagmaPlayer[0];
    }
}

File: magma-platform/magma-platform-bukkit/src/main/kotlin/de/jalumu/magma/platform/bukkit/player/BukkitPlayerProvider.kt
package de.jalumu.magma.platform.bukkit.player

import de.jalumu.magma.platform.base.player.BasePlayerProvider
import de.jalumu.magma.platform.bukkit.bootstrap.MagmaBukkitBootstrap
import de.jalumu.magma.player.MagmaPlayer
import org.bukkit.entity.Player
import java.util.*

class BukkitPlayerProvider(private val magma: MagmaBukkitBootstrap) : BasePlayerProvider() {
    override fun getOnlinePlayer(name: String): MagmaPlayer {
        val player = magma.server.getPlayer(name)
        return BukkitPlayer(magma, player!!)
    }

    override fun getOnlinePlayer(uuid: UUID): MagmaPlayer {
        val player = magma.server.getPlayer(uuid)
        return BukkitPlayer(magma, player!!)
    }

    override fun getOnlinePlayers(): Array<MagmaPlayer> {
        return magma.server.onlinePlayers.stream().map { player: Player? -> BukkitPlayer(magma, player!!) }.toArray() as Array<MagmaPlayer>
    }
}